src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -170,9 +170,8 @@ private static void HtmlDecode(string value, StringBuilder output)
                     int index = value.IndexOfAny(s_htmlEntityEndingChars, i + 1);
                     if (index > 0 && value[index] == ';')
                     {
-                        string entity = value.Substring(i + 1, index - i - 1);
-
-                        if (entity.Length > 1 && entity[0] == '#')
+                        int entityLength = index - i - 1;
+                        if (entityLength > 1 && value[i + 1] == '#')
```

Comment:
@jamesqo, your change based on my comment seems to have disappeared when you squashed.